File: MWORKS高校星火计划资料包/培训课程配套材料/01-官网课程配套材料/01-数学与基础应用/03-Julia语法-入门/附件示例/对应课程PPT页码/05_核心风格/page_178/Julia&python/kurtosis.jl
# Julia 的 kurtosis 实现
using Statistics
function kurtosis(X)
    mu = mean(X)
    n = length(X)
    tmp = (X .- mu) .^ 2
    # 对 tmp 中的每个元素求平方并求和，它等价于 sum(tmp.^2) 但是会更高效
    fourth_moment = sum(abs2, tmp) / n
    second_moment = sum(tmp) / n
    return fourth_moment / (second_moment^2)
end

File: references/MWORKS/MWORKS高校星火计划资料包/培训课程配套材料/01-官网课程配套材料/01-数学与基础应用/03-Julia语法-入门/附件示例/对应课程PPT页码/05_核心风格/page_178/Julia&python/kurtosis.py
#  使用 Python numpy 的 kurtosis 实现
import numpy as np

def kurtosis(X):
    mu = np.mean(X)
    n = len(X)
    tmp = (X - mu)**2
    fourth_moment = np.sum(tmp**2) / n
    second_moment = np.sum(tmp) / n
    return fourth_moment / (second_moment**2)



